doctr/models/core.py

```diff
@@ -41,6 +41,10 @@ def __call__(
         **kwargs: Any,
     ) -> List[Document]:
 
+        # Dimension check
+        if any(page.ndim != 3 for doc in documents for page in doc):
+            raise ValueError("incorrect input shape: all documents are expected to be list of multi-channel 2D images.")
```

Comment:
We should handle the 1 channel case

Reply:
at least a duplication of the channel thrice to simulate a RBG image

Reply:
Not sure I follow? I'm checking the number of dimensions / axes here, not the number of channels on the channel axis. Or are you suggesting to add a channel check? (if so, it depends on the architecture, we can easily imagine a model using grayscale)

Reply:
ok sorry I thought you were checking the number of channel on the channel axis